libdispatch/dinfermodel.c

```diff
@@ -1288,7 +1289,8 @@ closemagic(struct MagicFile* file)
 #ifdef USE_PARALLEL
         if (file->use_parallel) {
 	    int retval;
-	    if((retval = MPI_File_close(&file->fh)) != MPI_SUCCESS)
+	    if(file->fh != MPI_FILE_NULL
+		&& (retval = MPI_File_close(&file->fh)) != MPI_SUCCESS)
```

Comment:
I think these are good changes. Did the code generate an MPI warning? Or is this preventative coding?

Reply:
This is a while ago, so I don't recall the specific circumstances, but there definitely is a valid code path that leads here with file->fh being NULL.